Sources/SwiftSyntax/Syntax.swift

```diff
@@ -454,10 +454,20 @@ extension Syntax {
   }
 }
 
+/// Determines if two nodes are equal to each other.
+func ==(lhs: _SyntaxBase, rhs: _SyntaxBase) -> Bool {
+  return lhs.data.nodeId == rhs.data.nodeId
+}
+
+/// Determines if two nodes are equal to each other.
+public func ==(lhs: Syntax, rhs: Syntax) -> Bool {
```

Comment:
These seem left-over from testing.

Reply:
Oh wait, no, this is to work around the fact that `Syntax` itself is not Equatable. Which is the right move.

Reply:
On a related note, should we expose `nodeId` from the `Syntax` API ? Otherwise you can't use a bare `Syntax` node as a dictionary key, you have to cast it to a concrete type.

Reply:
> you have to cast it to a concrete type

To clarify, that was the case before any of these changes.

Reply:
I don't think it would hurt. Let's make sure the properties are `internal`, and maybe rename it to something like `identifier` or `uniqueIdentifier`.